### ISSUE TITLE
Be careful if you are using this package!

### ISSUE THREAD

**Issue by alosha79** [christophM/rulefit#27]
Here is an example of why:

```
X = X.matrix()
rf.fit(X, y, feature_names=features)
```
The function 'fit' accepts a matrix which converts you integers to floats. Then if you have a column with integer value, for example, gender (0, 1) then the code will create the rules that don't make sense. For instance, it will create a rule 'when gender  > 0.5'. But it should create 'when gender  == 1' or 'when gender == 0'.

**Comment by christophM** [comment]
That's common behaviour of decision trees and random forests for binary trees.
I agree it would be nicer to directly encode binary features as factors and not as numeric feature.

I'd argue that if you know how your data is encoded (e.g. gender 0 = male, gender 1 =female) and how trees work, that you will correctly interpret gender > 0.5 as "if female"

But it would be nice to directly encode the binary feature as a factor.
Do you know of a good solution for treating factors in Python?

**Comment by lukauskas** [comment]
Currently there's no good solution for categorical data in decision trees.
`sklearn` is working on something (see [here](https://github.com/scikit-learn/scikit-learn/pull/4899)) but so far it hasn't been merged.

It is generally accepted that one-hot encoding and the >0.5 or <= 0.5 splits will produce a comparable model in terms of prediction accuracy, but maybe not as interpretable.

**Comment by alosha79** [comment]
One possible solution for this package is to pass X as a data frame but not as a matrix. Whenever you need to access the column, then instead 
`X[:, i_col]`
you could use
`X.iloc[:, icol]`
This will pass integers. `sklearn` accepts dataframes so you package would be consistent with it. Then it would be `sklearn`'s responsibility how to treat integers.

**Comment by lukauskas** [comment]
These 0.5 splits come from `sklearn` not from `rulefit`. 
It's the implementation of their trees that create the floats.

As far as I am aware `sklearn` does not work with `DataFrames`, they even try to avoid the extra dependancy on `pandas`, [see the readme](https://github.com/scikit-learn/scikit-learn). The reason why dataframes 'work' in `sklearn` is that they convert it to `numpy` arrays behind the scenes (lists of lists also work because of that). At any rate the `integer` datatype is stored in the `numpy` array metadata (dataframes are just numpy arrays with index, anyway).

In practice this is a non-issue for binary variables once you convince yourself that `x > 0.5`  is true only when `x==1`. Some postprocessing on the resulting rules might be needed for visualisation, but as far as the ML task the representation is equivalent.

It's only a concern for non-ordinal factors (e.g, `weather = {'sunny', 'snowy', 'rainy'}`). In this case you would need to do one-hot encoding and this is when some proper handling of categorical variables might be better. But to be fair, you would already need to have your data encoded for the 'linear' bit of `rulefit` anyway as regression does not work with categories either.

Unless there is some categorical random tree implementation in python that I am not aware of, I think this is better discused in `sklearn` repo, for instance [Issue 5442](https://github.com/scikit-learn/scikit-learn/issues/5442), [PR 4899](https://github.com/scikit-learn/scikit-learn/pull/4899) as there is little that can be done here besides maybe a FAQ point in the documentation.
